### ISSUE TITLE
File Formats with multiple possible magic values

### ISSUE THREAD

**Issue by Keno** [JuliaIO/FileIO.jl#28]
I wanted to add Object File formats, but ran into the problem that MachO can use several magic numbers, which doesn't currently seem supported. Here's what I tried:

```julia
# Object File Formats
add_format(format"ELF", UInt8['\177','E','L','F'], [".o",".a",".so",""])
add_loader(format"ELF", :ELF)
add_saver(format"ELF", :ELF)

macho_magics = [0xfeedface,0xfeedfacf,0xcafebabe]
append!(macho_magics,map(bswap,macho_magics))
add_format(format"MachO", map(x->reinterpret(UInt8, [x]),macho_magics), [".o",".a",".dylib",""])
add_loader(format"MachO", :MachO)
add_saver(format"MachO", :MachO)

add_format(format"ELF", UInt8['M','Z'], [".o",".a",".lib",".obj",".exe"])
add_loader(format"ELF", :COFF)
add_saver(format"ELF", :COFF)
```

**Comment by SimonDanisch** [comment]
It should be relatively easy to add. Until that's added, you could just register a function that checks the different magic numbers.

**Comment by SimonDanisch** [comment]
fixed by #31
